notebooks/test.sh

```diff
@@ -6,23 +6,46 @@ set -x
 
 # path to the notebook/ directory.
 DIR=`dirname "${BASH_SOURCE[0]}"`
+# Install learntools branch
+pip3 install $DIR/..
+# The learntools repo is cloned to a read-only location. Various testing steps involve writing,
+# so copy the whole notebooks directory to a writeable location and work from there.
+WORKING_DIR=`mktemp -d`
+cp -r $DIR $WORKING_DIR
+cd $WORKING_DIR/notebooks
+
 TMP_DIR=`mktemp -d`
 
 # Install packages the notebook pipeline depends on but which aren't installed with the learntools package.
-pip3 install -r "$DIR/requirements.txt"
-
-# TRACKS="embeddings pandas python"
-# for track in $TRACKS
-# do
-#     # Run each step of the rendering pipeline, to make sure it runs without errors.
-#     python3 clean.py $track
-#     python3 prepare_push.py $track
-#     python3 render.py $track
+pip3 install -r requirements.txt
 
-#     # TODO: Once basic checks are working we should also do the following here:
-#     # - download necessary datasets for this track
-#     # - nbconvert execute *all* exercise notebooks for the track
-# done
+TRACKS="embeddings pandas python"
+for track in $TRACKS
+do
+    # Run each step of the rendering pipeline, to make sure it runs without errors.
+    python3 clean.py $track
+    python3 prepare_push.py $track
+    # TODO: Fails due to failure when calling get_git_branch()
+    #python3 render.py $track
+done
 
-# For now, just run one notebook (which doesn't depend on any datasets)
-jupyter nbconvert --output "$TMP_DIR" --execute "$DIR/python/raw/ex_2.ipynb"
+TESTABLE_NOTEBOOK_TRACKS="python pandas"
+for track in $TESTABLE_NOTEBOOK_TRACKS
```

Comment:
For my knowledge, what are the most relevant conditions needed for a track to be testable?

Just that it has tagged anywhere we expect it to raise exceptions and it has setup_data.sh?

Reply:
Yup, exactly.